PORTAL/jobs/__main__.py

```diff
@@ -400,92 +403,96 @@ def cmd_run_next(jobs: Jobs) -> None:
-    paused = _queue.paused
-    pid, pid_running = _queue.locked
-    if withlog:
-        log = list(_queue.read_log())
-
-    print('Job Queue:')
-    print(f'  size:     {len(queued)}')
-    #print(f'  max size: {maxsize}')
-    print(f'  paused:   {paused}')
-    if isinstance(pid, str):
-        assert pid_running is None, repr(pid_running)
-        print(f'  lock:     bad PID file (content: {pid!r})')
-    elif pid:
-        running = '' if pid_running else ' (not running)'
-        print(f'  lock:     held by process {pid}{running}')
-    else:
-        print('  lock:     (not locked)')
-    print()
-    print('Files:')
-    print(f'  data:      {render_file(_queue.datafile)}')
-    print(f'  lock:      {render_file(_queue.lockfile)}')
-    print(f'  log:       {render_file(_queue.logfile)}')
-    print()
-    print('Top 5:')
-    if queued:
-        for i in range(min(5, len(queued))):
-            print(f'  {i+1} {queued[i]}')
-    else:
-        print('  (queue is empty)')
-    if withlog:
+    for queue in jobs.iter_queues():
+        _queue = queue.snapshot
+        queued = _queue.jobs
+        paused = _queue.paused
+        pid, pid_running = _queue.locked
+        if withlog:
+            log = list(_queue.read_log())
+
+        print(f'Job Queue ({_queue.workerid}):')
+        print(f'  size:     {len(queued)}')
+        #print(f'  max size: {maxsize}')
+        print(f'  paused:   {paused}')
+        if isinstance(pid, str):
+            assert pid_running is None, repr(pid_running)
+            print(f'  lock:     bad PID file (content: {pid!r})')
+        elif pid:
+            running = '' if pid_running else ' (not running)'
+            print(f'  lock:     held by process {pid}{running}')
+        else:
+            print('  lock:     (not locked)')
         print()
-        print(f'Log size:    {len(log)}')
-        print('Last log entry:')
-        if log:
-            print('-'*30)
-            print()
-            for line in log[-1].render():
-                print(line)
-            print()
-            print('-'*30)
+        print('Files:')
+        print(f'  data:      {render_file(_queue.datafile)}')
+        print(f'  lock:      {render_file(_queue.lockfile)}')
+        print(f'  log:       {render_file(_queue.logfile)}')
+        print()
+        print('Top 5:')
+        if queued:
+            for i in range(min(5, len(queued))):
+                print(f'  {i+1} {queued[i]}')
         else:
-            print('  (log is empty)')
+            print('  (queue is empty)')
+        if withlog:
+            print()
+            print(f'Log size:    {len(log)}')
+            print('Last log entry:')
+            if log:
+                print('-'*30)
+                print()
+                for line in log[-1].render():
+                    print(line)
+                print()
+                print('-'*30)
+            else:
+                print('  (log is empty)')
 
 
 def cmd_queue_list(jobs: Jobs) -> None:
```

Comment:
I was going to suggest the option to specify which queue to list.  It isn't critical for this PR though.

Reply:
Done